src/upgrades/pre_0_14_3/mod.rs

```diff
@@ -0,0 +1,262 @@
+            // new store.
+            match migrate_agg_init::<A>(&handle, &cur_kv, &new_kv)? {
+                None => continue, // Cannot migrate this one.
+                Some(mut agg) => {
+                    // Apply and migrate each next command. This will error out
+                    // only if we cannot store the next command in the new store.
+                    while migrate_next_agg_command(&handle, &mut agg, &cur_kv, &new_kv)? {
+                        debug!("migrated command for {handle}, now at version: {}", agg.version());
+                    }
+                    let snapshot_key = AggregateStore::<A>::key_for_snapshot(&handle);
+                    new_kv.store(&snapshot_key, &agg)?;
+                    info!("Migrated {handle} in store {name}");
+                }
+            }
+        }
+
+        Ok(true)
+    }
+}
+
+/// This function is used to recover from corrupt command files resulting
+/// from issue #1160 - if needed for Write Ahead Log (WAL) stores.
+///
+/// In case the store has no apparent issues in "warming" the entities, then
+/// no migration is started.
+///
+/// If any issue is found, then we replay all commands for each entity (aggregate)
+/// starting from the last snapshot, in as far as they can be parsed and applied,
+/// into new migrated aggregate store.
+pub fn upgrade_wal<W: WalSupport>(name: &Namespace, config: &Config) -> UpgradeResult<bool> {
+    let current_wal_store = WalStore::<W>::create(&config.storage_uri, name)?;
+    if current_wal_store.warm().is_ok() {
+        info!("No need to migrate store {name}");
+        Ok(false)
+    } else {
+        info!("Issue found in store {name}. Will attempt to fix it in upgrade.");
+
+        let cur_kv = KeyValueStore::create(&config.storage_uri, name)?;
+        let new_kv = KeyValueStore::create_upgrade_store(&config.storage_uri, name)?;
+
+        for handle in current_wal_store.list()? {
+            // Try to get an initialised wal support type. This can only error out in
+            // case there is a valid snapshot but it cannot be saved in the new store.
+            match migrate_wal_snapshot::<W>(&handle, &cur_kv, &new_kv)? {
+                None => continue, // Cannot migrate this one.
+                Some(mut wal) => {
+                    // Apply and migrate each next command. This will error out
+                    // only if we cannot store the next command in the new store.
+                    while migrate_next_wal_command(&handle, &mut wal, &cur_kv, &new_kv)? {
+                        debug!("migrated change set for {handle}, now at version: {}", wal.revision());
+                    }
+                    info!("Migrated {handle} in store {name}");
+                }
+            }
+        }
+
+        Ok(true)
+    }
+}
+
+/// Check the task store for corrupted tasks due to issue #1160, and if present
+/// drop them in place. This is safe to do because missing tasks are re-added at start up.
```

Comment:
Why, here and in the warning messages below, do you use the term "drop"? "Drop" has specific meaning in Rust but that isn't the meaning that applies here, rather items are being removed/deleted from the store, and in the most common (currently the only) end-user scenarion this will be deletion of files on disk. Wouldn't delete be a word that operators will understand better than drop?